src/api/client.ts

```diff
@@ -0,0 +1,29 @@
+import axios from "axios";
+import router from "./../router";
+
+import { useUserStore } from "@/stores/user";
+
+export const defaultClient = axios.create({
+  baseURL: "http://localhost:3000/",
```

Comment:
You usually want to have this set globally in the .env file which you'd import like import.meta.env.ENV-NAME
https://vitejs.dev/guide/env-and-mode.html#env-files

Reply:
I briefly tried this but faced issues when it was importing the path both. On to do list as I know it is bad practice to have this info public